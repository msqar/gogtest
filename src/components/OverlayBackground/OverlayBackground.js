import React from 'react';
import { EventEmitter } from '../../services/event-emitter';
import { Events } from '../../services/index';
import { throttle } from 'lodash';

class OverlayBackground extends React.PureComponent {

    constructor(props) {
        super(props);

        this.state = {
            isVisible: false
        }
    }

    componentDidMount() {
        EventEmitter.subscribe(Events.DROPDOWN_TOGGLE_EVENT, this.onDropdownToggle);
    }

    onDropdownToggle = ({isVisible}) => {
        this.setState({isVisible});
    }

    onOverlayClick = () => {
        this.setState({isVisible: false});
        EventEmitter.dispatch(Events.OVERLAY_CLICKED_EVENT, {});
    }

    render() {
        const { isVisible } = this.state;
        return (
            <>
                <div className={'OverlayBackground ' + (isVisible ? 'is--visible' : '')} onClick={ throttle(this.onOverlayClick, 300) }/>
            </>
        )
    }
}

export default OverlayBackground;