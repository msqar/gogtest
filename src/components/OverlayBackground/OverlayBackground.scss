.OverlayBackground {
    position: fixed;
    width: 100vw;
    height: 100vh;
    z-index: 10;
    left: 0;
    top: 0;
    background-color: $gray-300;
    transition: opacity 300ms, visibility 300ms;
    opacity: 0;
    visibility: hidden;

    &.is--visible {
        visibility: visible;
        opacity: 0.5;
    }
}