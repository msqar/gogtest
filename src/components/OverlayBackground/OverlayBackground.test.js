import React from 'react';
import renderer from 'react-test-renderer';
import OverlayBackground from './OverlayBackground';

it('renders OverlayBackground correctly', () => {
    const tree = renderer.create(<OverlayBackground />).toJSON();
    expect(tree).toMatchSnapshot();
});