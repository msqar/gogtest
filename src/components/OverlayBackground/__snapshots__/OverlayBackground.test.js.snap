// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders OverlayBackground correctly 1`] = `
<div
  className="OverlayBackground "
  onClick={[Function]}
/>
`;
